src/lib/constants.ts

```diff
@@ -1,43 +1,53 @@
 import * as path from 'path';
 import { checkString } from './validation';
 
-const bootMountPointFromEnv = checkString(process.env.BOOT_MOUNTPOINT);
-const rootMountPoint = checkString(process.env.ROOT_MOUNTPOINT) || '/mnt/root';
-
 const supervisorNetworkInterface = 'supervisor0';
 
+// /mnt/root is the legacy root mountpoint
+const rootMountPoint = checkString(process.env.ROOT_MOUNTPOINT) || '/mnt/root';
+const withRootMount = (p: string) => path.join(rootMountPoint, p);
+const bootMountPoint =
+	checkString(process.env.BOOT_MOUNTPOINT) || withRootMount('/mnt/boot');
```

Comment:
should this be `/mnt/boot` instead of `withRootMount('/mnt/boot')`

Reply:
I updated boot, data, and state mountpoints to not use `withRootMount` as fallback [here](https://github.com/balena-os/balena-supervisor/pull/2120/files#diff-8c943ccfb5928fff63708e99eaafa93f8845c9405ed8e4b32afed23e5003025aR9-R12)